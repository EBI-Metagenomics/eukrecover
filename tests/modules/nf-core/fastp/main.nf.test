nextflow_process {

    name "Test Process FASTP"
    script "modules/nf-core/fastp/main.nf"
    process "FASTP"
    tag "modules"
    tag "modules_nfcore"
    tag "fastp"

    test("paired end") {

        when {

            process {
                """
                input[0] = tuple([ id:'test_pe', erz: "ERZ1" ],
                    [file(params.test_data['sarscov2']['illumina']['fastq_pe_1'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['fastq_pe_2'], checkIfExists: true)])
                input[1] = []
                input[2] = false
                input[3] = false
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
    test("single end") {

        when {

            process {
                """
                input[0] = tuple([ id:'test_se', erz: "ERZ1" ],
                    [file(params.test_data['sarscov2']['illumina']['fastq_se'], checkIfExists: true)])
                input[1] = []
                input[2] = false
                input[3] = false
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
