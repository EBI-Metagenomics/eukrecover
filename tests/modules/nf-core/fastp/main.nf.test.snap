{
    "paired end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        "test_pe.fastp.json:md5,d5ab7ce83e7443ff9a5d3fa2038f1f0d"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        "test_pe.fastp.html:md5,f2f1958d9455dd3af5ae458187d756e0"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        "test_pe.fastp.log:md5,b1729daaee4a39cd4c182ac115f0f55c"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        [
                            "test_pe_1.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd",
                            "test_pe_2.fastp.fastq.gz:md5,e59d0d65cc8c2801f1b970df601fc207"
                        ]
                    ]
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    "versions.yml:md5,48ffc994212fb1fc9f83a74fa69c9f02"
                ],
                "html": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        "test_pe.fastp.html:md5,f2f1958d9455dd3af5ae458187d756e0"
                    ]
                ],
                "json": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        "test_pe.fastp.json:md5,d5ab7ce83e7443ff9a5d3fa2038f1f0d"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        "test_pe.fastp.log:md5,b1729daaee4a39cd4c182ac115f0f55c"
                    ]
                ],
                "reads": [
                    [
                        {
                            "id": "test_pe",
                            "erz": "ERZ1"
                        },
                        [
                            "test_pe_1.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd",
                            "test_pe_2.fastp.fastq.gz:md5,e59d0d65cc8c2801f1b970df601fc207"
                        ]
                    ]
                ],
                "reads_fail": [
                    
                ],
                "reads_merged": [
                    
                ],
                "versions": [
                    "versions.yml:md5,48ffc994212fb1fc9f83a74fa69c9f02"
                ]
            }
        ],
        "timestamp": "2023-11-08T13:26:06.165728"
    },
    "single end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.json:md5,bfa29b0a17501eb734e51337afc85244"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.html:md5,c8618bea9ab961ec1d170dc8f89ba832"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.log:md5,7a153ed4327536f144cac8bec952b47e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd"
                    ]
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    "versions.yml:md5,48ffc994212fb1fc9f83a74fa69c9f02"
                ],
                "html": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.html:md5,c8618bea9ab961ec1d170dc8f89ba832"
                    ]
                ],
                "json": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.json:md5,bfa29b0a17501eb734e51337afc85244"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.log:md5,7a153ed4327536f144cac8bec952b47e"
                    ]
                ],
                "reads": [
                    [
                        {
                            "id": "test_se",
                            "erz": "ERZ1"
                        },
                        "test_se.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd"
                    ]
                ],
                "reads_fail": [
                    
                ],
                "reads_merged": [
                    
                ],
                "versions": [
                    "versions.yml:md5,48ffc994212fb1fc9f83a74fa69c9f02"
                ]
            }
        ],
        "timestamp": "2023-11-08T13:26:19.322639"
    }
}